elastic/beats
# use ADC on GCP instances

## What does this PR do?

Instead of requiring a file with credentials, the beat uses Google's ADC



## Why is it important?



Without this change, Filebeat requires a path to a file with credentials to be provided in its config and is unable to use for example a default service account.

## Checklist




## Author's Checklist



## How to test this PR locally



1. Upload the Filebeat binary to a GCP instance created with default config
1. Run the binary configured to read from PubSub, but do not provide any credentials
1. Filebeat should be able to detect that it's on GCP and use ADC (get a token for the default service account for the VM from the metadata server)

## Related issues


- Closes https://github.com/elastic/beats/issues/15667

## Use cases



Sending logs to Elastic from PubSub queue

## Screenshots



## Logs
